src/bluesky/utils/__init__.py

```diff
@@ -1868,3 +1869,43 @@ async def maybe_await(ret: SyncOrAsync[T]) -> T:
         # Mypy does not understand how to narrow type to non-awaitable in this
         # instance, see https://github.com/python/mypy/issues/15520
         return ret  # type: ignore
+
+
+class Plan:
+    def __init__(self, f, *args, **kwargs) -> None:
+        self._iter = f(*args, **kwargs)
+        self._stack = traceback.format_stack()
+        # Find the index in the stack trace where the function `f` is called
+        func_index = next((i for i, s in enumerate(self._stack) if f.__name__ in s), None)
+        if func_index is not None:
+            self._stack = self._stack[func_index:func_index+1]
```

Comment:
```suggestion
        self._stack = self._stack[:-2]
```

Talking about this in the hackathon room we came to consusus that accepting very long tracebacks to start to err on the side of giving context of "why was I creating this un-iterated plan" not just the exact code that created it.

We have a very old issue about generally filtering tracebacks for RE exceptions (no one wants to see the N levels of locally defined closure plans made by plan_mutator!) and filtering these tracebacks and those tracebacks should probably follow the same rules.  Thus, we should put off the filtering to do those togethor (or until we get real world feedback that verbose warnings are completely untenable.

The `-2` is to chop off the wrapper and `Plan.__init__` layers.

Reply:
Seems like I have a knack for leaving the zoom calls just before this PR gets discussed 😆 
Removed filtering in https://github.com/bluesky/bluesky/pull/1709/commits/88ccaf84a296dcf7f958f041728bcba7ae4d39cb

Reply:
You can use this power to force discussion on your PRs 🙃